zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java

```diff
@@ -1920,6 +1920,25 @@ private void getNoteByRevision(NotebookSocket conn, Notebook notebook, Message f
             .put("note", revisionNote)));
   }
 
+  private void getNoteByRevisionForCompare(NotebookSocket conn, Notebook notebook,
+      Message fromMessage) throws IOException {
+    String noteId = (String) fromMessage.get("noteId");
+    String revisionId = (String) fromMessage.get("revisionId");
+
+    String position = (String) fromMessage.get("position");
+    AuthenticationInfo subject = new AuthenticationInfo(fromMessage.principal);
+    Note revisionNote;
+    if (revisionId.equals("Head")) {
+      revisionNote = notebook.loadNoteFromRepo(noteId, subject);
```

Comment:
for this case, i think it's better to use `notebook.getNote(noteId)` since the latest note will be in memory always